src/main/java/com/impactupgrade/nucleus/service/segment/XeroAccountingPlatformService.java

```diff
@@ -425,30 +443,76 @@ protected String getExceptionDetails(Exception e) {
+//        if (crmContact.crmRawObject instanceof SObject sObject) {
+//            String supporterId = (String) sObject.getField(SUPPORTER_ID_FIELD_NAME);
+//            contact.setAccountNumber(supporterId);
+//        }
+//
+//        // TODO: temp
+//        env.logJobInfo("contact {} {} {} {} {} {}", crmContact.id, contact.getFirstName(), contact.getLastName(), contact.getName(), contact.getEmailAddress(), contact.getAccountNumber());
+//
+//        return contact;
+//    }
+
     protected Contact toContact(CrmContact crmContact) {
         if (crmContact == null) {
             return null;
         }
+
+        String supporterId;
+        if (crmContact.crmRawObject instanceof SObject sObject) {
+            supporterId = (String) sObject.getField(SUPPORTER_ID_FIELD_NAME);
+        } else {
+            //Should be unreachable
+            supporterId = crmContact.account.id;
+        }
+
         Contact contact = new Contact();
-        contact.setFirstName(crmContact.firstName);
-        contact.setLastName(crmContact.lastName);
-        contact.setName(crmContact.getFullName());
+        contact.accountNumber(supporterId);
+        Phone mobilePhone = new Phone();
+        mobilePhone.setPhoneType(Phone.PhoneTypeEnum.MOBILE);
+        mobilePhone.setPhoneNumber(crmContact.mobilePhone);
+        //TODO: area/country codes?
+        contact.setPhones(List.of(mobilePhone)); //TODO: add home/work?
         contact.setEmailAddress(crmContact.email);
-        if (!Strings.isNullOrEmpty(crmContact.mailingAddress.street)) {
-            Address address = new Address()
-                    .addressLine1(crmContact.mailingAddress.street)
-                    .city(crmContact.mailingAddress.city)
-                    .region(crmContact.mailingAddress.state)
-                    .postalCode(crmContact.mailingAddress.postalCode)
-                    .country(crmContact.mailingAddress.country);
-            contact.setAddresses(List.of(address));
+        if (crmContact.account.billingAddress != null) {
+            contact.setAddresses(List.of(toAddress(crmContact.account.billingAddress)));
         }
 
-        // TODO: make this part not-sfdc specific?
-        // TODO: SUPPORTER_ID_FIELD_NAME is DR specific
-        if (crmContact.crmRawObject instanceof SObject sObject) {
-            String supporterId = (String) sObject.getField(SUPPORTER_ID_FIELD_NAME);
-            contact.setAccountNumber(supporterId);
+        if (crmContact.account.recordType == EnvironmentConfig.AccountType.HOUSEHOLD) {
+            // Household
+            contact.setName(crmContact.getFullName() + " " + supporterId);
+            contact.setFirstName(crmContact.firstName);
+            contact.setLastName(crmContact.lastName);
+        } else {
+            // Organization
+            //TODO: Three different record types to include: AU ORGANISATION, AU CHURCH, AU SCHOOL?
+            contact.setName(crmContact.account.name + " " + supporterId);
```

Comment:
@VSydor Look through the doc I put together and DR AU's data mapping sheet to verify this one. Are organization record types supposed to use supporterId, or was supporterId for household contacts only and organizations simply use accountId?

Reply:
Will do today!

Reply:
Changed to crmContact.account.name + " " + accountNumber (accountId for orgs)